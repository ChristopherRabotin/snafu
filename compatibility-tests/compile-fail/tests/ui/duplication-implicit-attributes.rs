use snafu::prelude::*;

#[derive(Debug)]
struct ImplicitData;

impl snafu::GenerateImplicitData for ImplicitData {
    fn generate() -> Self {
        Self
    }
}

#[derive(Debug, Snafu)]
enum EnumError {
    AVariant {
        // Second attribute should be marked as duplicate
        #[snafu(implicit)]
        #[snafu(implicit)]
        my_data: ImplicitData,
    },
}

fn main() {}
