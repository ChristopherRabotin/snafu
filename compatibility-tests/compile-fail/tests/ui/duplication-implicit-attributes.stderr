error: Multiple `implicit` attributes are not supported on a field
  --> $DIR/duplication-implicit-attributes.rs:17:17
   |
17 |         #[snafu(implicit)]
   |                 ^^^^^^^^
